### PR TITLE
Update to Android SDK 0.12.0

### PR DESCRIPTION
**Pull request by PeterYan-UW**
Add search function
Add loading screen

### CODE REVIEW

File: singlevenue/src/main/java/com/mappedin/examples/singlevenue/MainActivity.java [hunk 1]
```diff
@@ -182,7 +196,41 @@ public void onClick(View v) {
...
+                    suggestListView.setVisibility(View.INVISIBLE);
+                    smartSearch.search(search.getText().toString(), 1000);
+                    return true;
+                }
+                return false;
+            }
+        });
+        search.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
```
Comment (csherratt) [inline singlevenue/src/main/java/com/mappedin/examples/singlevenue/MainActivity.java]: if these are no-ops, can we lave them blank?

File: singlevenue/src/main/java/com/mappedin/examples/singlevenue/MainActivity.java [hunk 2]
```diff
@@ -182,7 +196,41 @@ public void onClick(View v) {
...
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int start, int before, int count) {
+                final String input = charSequence.toString();
+                smartSearch.suggest(input, 500);
+            }
+
+            @Override
+            public void afterTextChanged(Editable editable) {
```
Comment (csherratt) [inline singlevenue/src/main/java/com/mappedin/examples/singlevenue/MainActivity.java]: ditto as above/

File: singlevenue/src/main/java/com/mappedin/examples/singlevenue/MainActivity.java [hunk 3]
```diff
@@ -182,7 +196,41 @@ public void onClick(View v) {
...
+        search.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int start, int before, int count) {
+                final String input = charSequence.toString();
+                smartSearch.suggest(input, 500);
```
Comment (csherratt) [inline singlevenue/src/main/java/com/mappedin/examples/singlevenue/MainActivity.java]: What does the 500 mean?

File: singlevenue/src/main/java/com/mappedin/examples/singlevenue/SearchCallback.java [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+                    new SuggestListAdapter(activity, suggests);
+            suggestListView.setAdapter(adapter);
+            suggestListView.setVisibility(View.VISIBLE);
+            suggestListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+                @Override
+                public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                    String selected = suggests[position];
+                    searchBar.setText(selected);
+                    suggestListView.setVisibility(View.INVISIBLE);
+                    smartSearch.search(selected, 1000);
```
Comment (csherratt) [inline singlevenue/src/main/java/com/mappedin/examples/singlevenue/SearchCallback.java]: What does the 1000 mean?

File: singlevenue/src/main/java/com/mappedin/examples/singlevenue/MainActivity.java (hunk 1 above)
Comment (PeterYan-UW) [reply singlevenue/src/main/java/com/mappedin/examples/singlevenue/MainActivity.java]: Couldn't do it in Java, it request me Override all method in that interface

File: singlevenue/src/main/java/com/mappedin/examples/singlevenue/MainActivity.java (hunk 2 above)
Comment (PeterYan-UW) [reply singlevenue/src/main/java/com/mappedin/examples/singlevenue/MainActivity.java]: same as above

File: singlevenue/src/main/java/com/mappedin/examples/singlevenue/SearchCallback.java (hunk 4 above)
Comment (PeterYan-UW) [reply singlevenue/src/main/java/com/mappedin/examples/singlevenue/SearchCallback.java]: add comment into code 1000 means 1000ms(1s)

File: singlevenue/src/main/java/com/mappedin/examples/singlevenue/MainActivity.java (hunk 3 above)
Comment (PeterYan-UW) [reply singlevenue/src/main/java/com/mappedin/examples/singlevenue/MainActivity.java]: add comment into code 500 means 500ms(0.5s)

**Pull request merged**